internal/permutations/permutations_test.go

```diff
@@ -0,0 +1,72 @@
+package permutations
+
+import "testing"
+
+const xxx = 10
+
+func TestModelSingle(t *testing.T) {
+	var seq, sent int64
+	var max int64 = xxx
+	// make a cache
+	var cache = make([]int64, xxx)
+	// create the permutations
+	p, _ := NewKensler(xxx)
+	// create the sequence to compare to
+	shuffle := genSequence(p)
+	// output
+	var output = make([]int64, xxx)
+	// input
+	var input = make([]int64, xxx)
+
+	// run it the same way the single threaded shuffler runs it
+	for i := 0; i < xxx; i++ {
+		input[i] = int64(i)
+		next := p.Shuffle(sent)
+
+		if next == seq {
+			//  we fall perfectly in sequence, write it out
+			output[sent] = int64(i)
+			sent++
+		} else {
+			// cache the current sequence
+			cache[seq] = int64(i)
+		}
+		seq++
+
+		if seq == max {
+			/*
+				for _, pos := range enc.permutations[enc.sent:] {
+					if _, err = enc.w.Write(enc.b[pos][:]); err != nil {
+						return
+					}
+				}
+			*/
```

Comment:
still needed?

Reply:
I guess not, its a template for the code just under